lib/datadog/appsec/contrib/rack/gateway/watcher.rb

```diff
@@ -114,10 +91,23 @@ def self.watch
                 [ret, res]
               end
             end
-            # rubocop:enable Metrics/AbcSize
-            # rubocop:enable Metrics/CyclomaticComplexity
             # rubocop:enable Metrics/MethodLength
-            # rubocop:enable Metrics/PerceivedComplexity
+
+            class << self
+              private
+
+              def active_trace
+                return unless defined?(Datadog::Tracing) && Datadog::Tracing.respond_to?(:active_trace)
+
+                Datadog::Tracing.active_trace
+              end
+
+              def active_span
+                return unless defined?(Datadog::Tracing) && Datadog::Tracing.respond_to?(:active_span)
+
+                Datadog::Tracing.active_span
+              end
+            end
```

Comment:
Yes it does, although only for data transport. I'm trying to keep `AppSec` as detached as possible from `Tracing`. Ideally it should be able to fully execute with tracing disabled, and only not report any data because of the lack of trace transport (FYI there is a blocking feature, currently unused: no rules in the current recommended ruleset is set to return a blocking decision. With tracing disabled AppSec should still be able to block).

I must admit I did not exercise that case lately and there may be gaps, but that's the general idea. Also event propagation in #1969 may pose an issue depending on the chosen implementation.

Reply:
Also I'm not sure but I think `&& Datadog::Tracing.respond_to?(:active_trace)` is not needed anymore. It was a carryover from `Datadog.active_root_span` which may not have been there in all cases but I believe `defined?(Datadog::Tracing)` is sufficient now.

Reply:
Interesting -- so would it be possible and a supported configuration to run AppSec with tracing disabled?

That would indeed change my observation, as in that case it would mean that even with tracing we'd want this class to do things.

Is that something we already support, or something that we're going to add later?

Reply:
We don't support it today but it was a requirement early on, back when the transport was separate the trace API and used a dedicated AppSec API.

Since I already wrote the code in a very isolated way from tracing, I find it both easier and beneficial to keep up with this separation of concerns as it would be hard to retrofit later on if coupling creeps in.

Reply:
If the goal is to move a Tracing-independent Appsec implementation, than I'm ok with the changes as is.